pkg/controller/reconciler/reconciler.go

```diff
@@ -25,6 +26,8 @@ import (
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
 )
 
+const componentFinalizer = "components.platform.opendatahub.io/finalizer"
```

Comment:
the `Reconciler` is now used also for services, and may be reused for other resources so maybe use something like:

```go
const platformFinalizer = "platform.opendatahub.io/finalizer"
```